### ISSUE TITLE
`event.name` field of the kubernetes events does not reflect events definition

### ISSUE THREAD

**Issue by tetianakravchenko** [open-telemetry/opentelemetry-collector-contrib#35857]
### Component(s)

receiver/k8sobjects

### Describe the issue you're reporting

receiver/k8sobjects seems to be a recommended way to collect kubernetes events, as there is an open discussion regarding the k8sevents deprecation - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24242:

```
receivers:
  k8sobjects:
    objects:
      - name: events
        mode: "watch"
        group: "events.k8s.io"
        exclude_watch_type:
          - "DELETED"
```

In this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16543 - `event.name` was set as a name of the kubernetes event.
For example, for the event:
```
- apiVersion: v1
  count: 1
  eventTime: null
  firstTimestamp: "2024-10-17T17:13:07Z"
  involvedObject:
    apiVersion: apps/v1
    kind: StatefulSet
    name: web
    namespace: opentelemetry-operator-system
    resourceVersion: "26119"
    uid: 664abece-c04c-484e-93c1-394d19dd200b
  kind: Event
  lastTimestamp: "2024-10-17T17:13:07Z"
  message: create Pod web-0 in StatefulSet web successful
  metadata:
    creationTimestamp: "2024-10-17T17:13:07Z"
    name: web.17ff4cc43c2c72a0
    namespace: opentelemetry-operator-system
    resourceVersion: "26122"
    uid: 7d7c3ea4-59f6-422e-be44-a489a0c058d8
  reason: SuccessfulCreate
  reportingComponent: statefulset-controller
  reportingInstance: ""
  source:
    component: statefulset-controller
  type: Normal
```

`event.name` is `web.17ff4cc43c2c72a0`, that is unique for each event.

as defined in https://opentelemetry.io/docs/specs/semconv/general/events/#event-definition:
```
The event name uniquely identifies event structure / type of the payload (body) and the set of attributes.
```
that contradicts with the current `event.name`

I found related issues, but for the k8sevents - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14474, that was closed. As was mentioned in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14474#issuecomment-1267729114:
```
Looking at the examples you've shown there, I'd say an appropriate event.name might be something like involvedObject.kind + reason
```

should `event.name` be changed to `involvedObject.kind` + `reason` ?

**Comment by TylerHelmuth** [comment]
@tetianakravchenko you are correct that we are currently not using the OTel Event definition for `event.name` correctly. We are also setting and event.domain (I think) which isn't spec-ed anymore.

This happened because k8sojects receiver was an early adopter of the Event concept as one of its prototypes for the spec. But now we are holding off on fixing it until Otel Events are stable so that we can do a single breaking change.

**Comment by tetianakravchenko** [comment]
@TylerHelmuth thank you for the reply!
> But now we are holding off on fixing it until Otel Events are stable so that we can do a single breaking change.

is there any related issue/s regarding the work on otel events that I can follow?
